plugin/path_config.go

```diff
@@ -169,6 +178,44 @@ func (b *GcpAuthBackend) pathConfigWrite(ctx context.Context, req *logical.Reque
 		return nil, fmt.Errorf("failed to persist configuration to storage: %w", err)
 	}
 
+	// Disable Automated Rotation and Deregister credentials if required
+	if c.DisableAutomatedRotation {
+		// Ensure de-registering only occurs on updates and if
+		// a credential has actually been registered (rotation_period or rotation_schedule is set)
+		deregisterReq := &rotation.RotationJobDeregisterRequest{
+			MountType: req.MountType,
+			ReqPath:   req.Path,
+		}
+		b.Logger().Debug("Deregistering rotation job", "mount", req.MountPoint+req.Path)
+		err := b.System().DeregisterRotationJob(ctx, deregisterReq)
+		if err != nil {
+			return logical.ErrorResponse("error de-registering rotation job: %s", err), nil
+		}
+	} else {
+		// Now that the root config is set up, register the rotation job if it's required.
+		if c.ShouldRegisterRotationJob() {
+			cfgReq := &rotation.RotationJobConfigureRequest{
+				Name:             rootRotationJobName,
+				MountType:        req.MountType,
+				ReqPath:          req.Path,
+				RotationSchedule: c.RotationSchedule,
+				RotationWindow:   c.RotationWindow,
+				RotationPeriod:   c.RotationPeriod,
+			}
+
+			_, err := rotation.ConfigureRotationJob(cfgReq)
+			if err != nil {
+				return logical.ErrorResponse("error configuring rotation job: %s", err), nil
+			}
```

Comment:
I believe this is not required anymore since RegisterRotationJob was recently changed to do this boilerplate.